### PR TITLE
remove mu_define_enum from repo code

### PR DESCRIPTION
**Pull request by danewalton**

### CODE REVIEW

**Comment by mamokarz** [comment]
> #include "azure_macro_utils/macro_utils.h"

Please remove

---
Refers to: inc/az_ulib_action_api.h:9 in 93e8224. [](commit_id = 93e8224018762c54f9f29a5e944604602903a837, deletion_comment = False)

**Comment by mamokarz** [comment]
> #include "umock_c/umock_c_prod.h"

Please remove

---
Refers to: inc/az_ulib_action_api.h:11 in 93e8224. [](commit_id = 93e8224018762c54f9f29a5e944604602903a837, deletion_comment = False)

**Comment by mamokarz** [comment]
> #include "umock_c/umock_c_prod.h"

Please remove

---
Refers to: inc/az_ulib_base.h:15 in 93e8224. [](commit_id = 93e8224018762c54f9f29a5e944604602903a837, deletion_comment = False)

**Comment by mamokarz** [comment]
> #include "umock_c/umock_c_prod.h"

Please remove

---
Refers to: inc/az_ulib_result.h:15 in 93e8224. [](commit_id = 93e8224018762c54f9f29a5e944604602903a837, deletion_comment = False)

**Comment by mamokarz** [comment]
> #include "azure_macro_utils/macro_utils.h"

Please remove

---
Refers to: inc/az_ulib_ulog.h:18 in 93e8224. [](commit_id = 93e8224018762c54f9f29a5e944604602903a837, deletion_comment = False)

File: inc/az_ulib_result.h [hunk 1]
```diff
@@ -27,8 +27,7 @@ extern "C" {
 /**
  * @brief   Enumeration of ulib results.
  */
-MU_DEFINE_ENUM(
-    AZ_ULIB_RESULT,
+typedef enum AZ_ULIB_RESULT_TAG {
```
Comment (mamokarz) [inline inc/az_ulib_result.h]: >AZ_ULIB_RESULT_TAG [](start = 13, length = 18)

It should be lowercase, you may fix or open an issue to fix it.

File: inc/az_ulib_ulog.h [hunk 2]
```diff
@@ -38,11 +38,10 @@ static const char* const AZ_ULOG_USTREAM_ILLEGAL_ARGUMENT_ERROR_STRING
 /**
  * @brief   enum to select the log type
  */
-MU_DEFINE_ENUM(
-    AZ_ULOG_TYPE,
+typedef enum AZ_ULOG_TYPE_TAG {
```
Comment (mamokarz) [inline inc/az_ulib_ulog.h]: >AZ_ULOG_TYPE_TAG [](start = 13, length = 16)

It should be lowercase, you may fix or open an issue to fix it.

File: inc/az_ulib_result.h (hunk 1 above)
Comment (danewalton) [reply inc/az_ulib_result.h]: yup i thought about fixing it here but i have anothe PR which makes all those changes next. decided to limit the scope here

**Pull request merged**